tests/test_math.py

```diff
@@ -185,10 +237,64 @@ def test_norm_float32(self):
         self.assert_allclose32(result, expected_val)
 
     def test_norm_float64(self):
-        cplx = mm.ComplexFloat64(self.real1_64, self.imag1_64)
+        cplx = mm.complex128(self.real1_64, self.imag1_64)
 
         result = cplx.norm()
 
         expected_val = self.real1_64 ** 2 + self.imag1_64 ** 2
 
         self.assert_allclose64(result, expected_val)
+
+    def test_dtype_verification_float32(self):
+        dtype = mm.complex64.dtype()
+        expected_dtype = np.dtype('complex64')
+
+        self.assertEqual(dtype, expected_dtype)
+
+    def test_dtype_verification_float64(self):
+        dtype = mm.complex128.dtype()
+        expected_dtype = np.dtype('complex128')
+
+        self.assertEqual(dtype, expected_dtype)
+
+    def test_complex_array_float32(self):
+        cplx = mm.complex64(self.real1_32, self.imag1_32)
+        sarr = mm.SimpleArrayComplex64(10)
+        sarr.fill(cplx)
+        ndarr = np.array(sarr, copy=False, dtype=mm.complex64.dtype())
+
+        for i in range(10):
+            self.assertEqual(ndarr[i].real, self.real1_32)
+            self.assertEqual(ndarr[i].imag, self.imag1_32)
+
+        self.assertEqual(ndarr.dtype, mm.complex64.dtype())
+
+        sarr = mm.SimpleArrayComplex64(array=ndarr)
```

Comment:
please also add tests for `mm.SimpleArray(..., dtype='complex128')` constructors.